app/models/user.rb

```diff
@@ -53,13 +53,15 @@ def admin?
   end
 
   def reputation(track_slug: nil, category: nil)
+    return super() if track_slug.nil? && category.nil?
```

Comment:
If you're explicitely checking for just `nil` and don't mind also checking for `false` you can do:
```
return super() unless track_slug || category
```

(I think my bool logic is right there!)

You might instead be checking to see if they're nil or blank/empty, in which case you'd be better using `.present?` or `.blank?` rather than `.nil?`

If you *really* care just about `nil` then what you have is fine, but that's unusual

Reply:
(In this context, I think you actually care about empty strings too, so present or blank)

Reply:
My original reasoning was that `nil` is the default value of the parameters, but I do indeed not really care about `nil` in particular.